kong/db/schema/typedefs.lua

```diff
@@ -75,6 +76,17 @@ local function validate_cidr_v4(ip)
 end
 
 
+local function validate_cidr_v4_v6(ip)
```

Comment:
How is this different from `local function validate_ip_or_cidr(ip)` above? I know it uses different library, but both accept `ip` and `cidr`. So could you just use the `typedefs.ip_or_cidr` with `ip-restriction` and skip adding this for now? We will rework these when removing other deps.

Reply:
we can break existing third party plugins using `validate_ip_or_cidr`

the new `ipmatcher` lib doesn't validate stuff like this:

```
{ whitelist = { "0.0.0.0/a", "0.0.0.0/-1", "0.0.0.0/33" }
```

i.e. `ipmatcher` doesn't complain on wrong cidrs... this issue has been reported at https://github.com/api7/lua-resty-ipmatcher/issues/11

we can also break the error message, but that is a very minor problem

Reply:
I don't think we break any existing plugins that use `validate_ip_or_cidr` as we ARE NOT CHANGING it. That typedefs.lua stays the same. This already catches invalid cidrs. We just change the ip-restriction plugin to use `validate_ip_or_cidr`. That validation function DOES NOT use `ipmatcher` but that mediador, and that is fine. The implementation inside `handler.lua` uses `ipmatcher`.